lms/static/js/discovery/discovery_factory.js

```diff
@@ -5,17 +5,23 @@
         'js/discovery/views/search_form', 'js/discovery/views/courses_listing',
         'js/discovery/views/filter_bar', 'js/discovery/views/refine_sidebar'],
         function(Backbone, SearchState, Filters, SearchForm, CoursesListing, FilterBar, RefineSidebar) {
-            return function(meanings, searchQuery) {
+            return function(meanings, searchQuery, userLanguage, userTimezone) {
                 var dispatcher = _.extend({}, Backbone.Events);
                 var search = new SearchState();
                 var filters = new Filters();
-                var listing = new CoursesListing({model: search.discovery});
                 var form = new SearchForm();
                 var filterBar = new FilterBar({collection: filters});
                 var refineSidebar = new RefineSidebar({
                     collection: search.discovery.facetOptions,
                     meanings: meanings
                 });
+                var listing;
+                var courseListingModel = search.discovery;
+                courseListingModel.userPreferences = {
+                    userLanguage: userLanguage,
+                    userTimezone: userTimezone
+                };
+                listing = new CoursesListing({model: courseListingModel});
```

Comment:
This seems to be the least invasive method to sneak the userPreferences var into the payload being sent through. It would be possible to go into the search.discovery method, but it seemed less messy to just attach it to the model here, and go forward. 
I'm a little reluctant to try to pass the user variables too much farther into this, as this is totally functional, though it's not perfectly beautiful.